api/service/version_endpoint_service.go

```diff
@@ -231,13 +231,10 @@ func (k *endpointService) override(left *models.VersionEndpoint, right *models.V
 		left.EnvVars = models.MergeEnvVars(left.EnvVars, right.EnvVars)
 	}
 
-	// default to HttpJson if not provided
-	if right.Protocol == "" {
-		right.Protocol = protocol.HttpJson
-	}
-
 	// override protocol
-	left.Protocol = right.Protocol
+	if right.Protocol != "" {
+		left.Protocol = right.Protocol
+	}
```

Comment:
This change is a partial reversion of the changes made in https://github.com/caraml-dev/merlin/pull/300 (change no. 2).

Reply:
That's a good point 🤔 I think after looking at the code, it seems like the only this `override` method is only [called](https://github.com/caraml-dev/merlin/blob/072325c6e1ae1226216db7186757b22a391f6c72/api/service/version_endpoint_service.go#L121) to update 2 types of endpoint configs (the `left` argument), 1) an existing endpoint config stored in the DB and 2) a 'default' endpoint config that will be overwritten by a user request. 

In case 2, we're actually already [setting](https://github.com/caraml-dev/merlin/blob/main/api/models/version_endpoint.go#L103) this `Protocol` field as `protocol.HttpJson` by default so `left.Protocol` wouldn't be empty in such cases.

For case 1, well, I don't think it's possible we'll encounter any empty strings for the `Protocol` field of the endpoints retrieved from the DB, if we assume that the mechanism mentioned in case 2 is working correctly 😅 Unless maybe there's some DB corruption or something similar 🤔

So yeah I don't really think it's necessary to include the extra step to set a default value for `left.Protocol` (it seems a little strange to set a default value in a method called 'override' also. What do you think?

Reply:
When we added the `protocol` column to version_endpoints and model_endpoints tables, we didn't specify the default value ([ref](https://github.com/caraml-dev/merlin/blob/main/db-migrations/29_protocol.up.sql)). So, it's possible that the `protocol` field is empty here, hence the `left.Protocol` would also be empty. I think it's better to set the default value for `left.Protocol` if it's empty.

Reply:
Got it, thanks for digging that up 😅 I've just added that additional step to set the default protocol too!